### PR TITLE
[#88] Docs changes for Domain

### PR DESCRIPTION
**Pull request by rahuldesirazu**

### CODE REVIEW

File: content/latest/api/ysql/commands/ddl_create_domain.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    parent: api-ysql-commands
+aliases:
+  - /latest/api/ysql/ddl_create_domain/
+isTocNested: true
+showAsideToc: true
+---
+
+## Synopsis
+
+The `CREATE DOMAIN` command creates a user-defined data type with a range, optional DEFAULT, NOT NULL and CHECK constraint.
```
Comment (ndeodhar) [inline content/latest/api/ysql/commands/ddl_create_domain.md]: It's not clear what "with a range" means here.
We should also add a line about why domains are needed - for example, "Domains are useful to abstract data types with common constraints. For example, domain can be used to represent phone number columns that will require the same CHECK constraints on the syntax."

File: content/latest/api/ysql/commands/ddl_create_domain.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+### Grammar
+```
+create_domain ::= CREATE DOMAIN name [ AS ] data_type
+    [ COLLATE collation ]
```
Comment (ndeodhar) [inline content/latest/api/ysql/commands/ddl_create_domain.md]: COLLATE is disallowed in our grammar, so we should remove this line.

File: content/latest/api/ysql/commands/ddl_create_domain.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+domain_constraint ::= [ CONSTRAINT constraint_name ]
+{ NOT NULL | NULL | CHECK (expression) };
+```
+
+Where
+
+- `name` is the name of the domain.
+- `data_type` is the underlying data type.
+- `COLLATE collation` is an optional collation for the domain.
+- `DEFAULT expression` sets default value for columns and must match data type of the domain.
```
Comment (ndeodhar) [inline content/latest/api/ysql/commands/ddl_create_domain.md]: s/for columns/for columns of the domain data type

File: content/latest/api/ysql/commands/ddl_create_domain.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+```sql
+postgres=# CREATE DOMAIN idx DEFAULT 5 CHECK (VALUE > 0);
+```
+
+```sql
+postgres=# CREATE TABLE sample(k text primary key, v idx);
+```
+
+## See Also
+[`DROP CONSTRAINT`](../ddl_drop_constraint)
```
Comment (ndeodhar) [inline content/latest/api/ysql/commands/ddl_create_domain.md]: ddl_drop_domain and ddl_alter_domain

File: content/latest/api/ysql/commands/ddl_create_domain.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+Where
+
+- `name` is the name of the domain.
+- `data_type` is the underlying data type.
+- `COLLATE collation` is an optional collation for the domain.
+- `DEFAULT expression` sets default value for columns and must match data type of the domain.
+- `CONSTRAINT constraint_name` is an optional name for constraint.
+- `NOT NULL` does not allow null values.
+- `NULL` allows null values (default).
+- `CHECK (expression)` must produce a boolean result.
```
Comment (ndeodhar) [inline content/latest/api/ysql/commands/ddl_create_domain.md]: Add that "This enforces a constraint that the values of the domain must satisfy. You should use the key word VALUE to refer to the value being tested. Expressions evaluating to TRUE or UNKNOWN succeed."

File: content/latest/api/ysql/commands/ddl_create_domain.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+## Semantics
+
+- An error is raised if `name` already exists in the specified database.
+
+## Examples
+
+```sql
+postgres=# CREATE DOMAIN idx DEFAULT 5 CHECK (VALUE > 0);
```
Comment (ndeodhar) [inline content/latest/api/ysql/commands/ddl_create_domain.md]: Personally, I would prefer a more useful example. Something like:
`CREATE DOMAIN phone_number AS TEXT CHECK(VALUE ~ '^\d{3}-\d{3}-\d{4}$');`
`CREATE TABLE person(first_name TEXT, last_name TEXT, phone_number phone_number);`

File: content/latest/api/ysql/commands/ddl_drop_domain.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+### Grammar
+```
+drop_domain ::= DROP DOMAIN [IF EXISTS ] name [, ...]  [ CASCADE | RESTRICT ]
+```
+
+Where
+
+- `IF EXISTS` does not throw an error if domain does not exist.
+- `name` is the name of the existing domain.
+- `CASCADE` automatically drops objects that depend on the domain.
```
Comment (ndeodhar) [inline content/latest/api/ysql/commands/ddl_drop_domain.md]: objects that depend on the domain such as table columns using the domain data type and, in turn, all other objects that depend on those objects.

File: content/latest/api/ysql/commands/ddl_drop_domain.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+```sql
+postgres=# DROP DOMAIN idx CASCADE;
+```
+
+## See Also
+
+[`CREATE DOMAIN`](../ddl_create_domain)
+[`ALTER DOMAIN`](../dml_alter_domain)
```
Comment (ndeodhar) [inline content/latest/api/ysql/commands/ddl_drop_domain.md]: ddl_alter_domain

**Review by ndeodhar — CHANGES_REQUESTED** [review]

File: content/latest/api/ysql/commands/ddl_create_domain.md (hunk 1 above)
Comment (nocaway) [reply content/latest/api/ysql/commands/ddl_create_domain.md]: Hi Rahul, it'd be clearer if you moved the word constraint up.
The command creates a user-defined datatype that has optional constraints such as range of valid values, NOT NULL, CHECK, and DEFAULT.

File: content/latest/api/ysql/commands/ddl_create_domain.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+domain_constraint ::= [ CONSTRAINT constraint_name ]
+{ NOT NULL | NULL | CHECK (expression) };
+```
+
+Where
+
+- `name` is the name of the domain.
+- `data_type` is the underlying data type.
+- `COLLATE collation` is an optional collation for the domain.
```
Comment (nocaway) [inline content/latest/api/ysql/commands/ddl_create_domain.md]: Remove collation option.

File: content/latest/api/ysql/commands/ddl_alter_domain.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+Where 
+
+- `SET/DROP DEFAULT` sets or removes the default value for a domain.
+- `RENAME` changes the name of the domain.
+- Other `ALTER DOMAIN` options are not yet supported.
+
+## Semantics
+
+- An error is raised if the specified `name` does not exist or `new_name` already exists.
```
Comment (nocaway) [inline content/latest/api/ysql/commands/ddl_alter_domain.md]: ... if DOMAIN 'name' does not exisit or DOMAIN 'new_name' already exists

File: content/latest/api/ysql/commands/ddl_alter_domain.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+Where 
+
+- `SET/DROP DEFAULT` sets or removes the default value for a domain.
+- `RENAME` changes the name of the domain.
+- Other `ALTER DOMAIN` options are not yet supported.
+
+## Semantics
+
+- An error is raised if DOMAIN `name` does not exist DOMAIN `new_name` already exists.
```
Comment (ndeodhar) [inline content/latest/api/ysql/commands/ddl_alter_domain.md]:
```suggestion
- An error is raised if DOMAIN `name` does not exist or DOMAIN `new_name` already exists.
```

File: content/latest/api/ysql/commands/ddl_create_domain.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    parent: api-ysql-commands
+aliases:
+  - /latest/api/ysql/ddl_create_domain/
+isTocNested: true
+showAsideToc: true
+---
+
+## Synopsis
+
+The `CREATE DOMAIN` command creates a user-defined data type with an optional constraints such as range of valid values, DEFAULT, NOT NULL and CHECK.
```
Comment (ndeodhar) [inline content/latest/api/ysql/commands/ddl_create_domain.md]:
```suggestion
The `CREATE DOMAIN` command creates a user-defined data type with optional constraints such as range of valid values, DEFAULT, NOT NULL and CHECK. 
```

File: content/latest/api/ysql/commands/ddl_create_domain.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+- `DEFAULT expression` sets default value for columns of the domain data type.
+- `CONSTRAINT constraint_name` is an optional name for constraint.
+- `NOT NULL` does not allow null values.
+- `NULL` allows null values (default).
+- `CHECK (expression)` enforces a constraint that the values of the domain must satisfy and returns a boolean value. 
+The key word VALUE should be used to refer to the value being tested. Expressions evaluating to TRUE or UNKNOWN succeed.
+
+## Semantics
+
+- An error is raised if `name` already exists in the specified database.
```
Comment (ndeodhar) [inline content/latest/api/ysql/commands/ddl_create_domain.md:56]:
```suggestion
- An error is raised if DOMAIN `name` already exists in the database.
```

File: content/latest/api/ysql/commands/ddl_drop_domain.md [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+Where
+
+- `IF EXISTS` does not throw an error if domain does not exist.
+- `name` is the name of the existing domain.
+- `CASCADE` automatically drops objects that depend on the domain such as table columns using the domain data type and, in turn, all other objects that depend on those objects.
+- `RESTRICT` refuses to drop the domain if objects depend on it (default).
+
+## Semantics
+
+- An error is raised if the specified name does not exist (unless `IF EXISTS` is set).
```
Comment (ndeodhar) [inline content/latest/api/ysql/commands/ddl_drop_domain.md]:
```suggestion
- An error is raised if the specified domain does not exist (unless `IF EXISTS` is set).
```

**Review by ndeodhar — COMMENTED** [review]
Thanks for making the changes.
Some minor comments, other than that LGTM.

**Pull request merged**